src/uv_append.c

```diff
@@ -258,8 +265,26 @@ static void uvAliveSegmentWriteCb(struct UvWriterReq *write, const int status)
 
 out:
     /* Fire the callbacks of all requests that were fulfilled with this
-     * write. */
-    uvAppendFinishWritingRequests(uv, status);
+     * write. Keep track of # of append entries that were rolled back, if any.
+     */
+    n = uvAppendFinishWritingRequests(uv, status);
+    if (status != 0) {
+        /* When the write has failed additionally cancel all future append
+         * related activity. This will also rewind uv->append_next_index. All
+         * append requests need to be canceled because raft assumes all appends
+         * happen in order and if an append fails (and is not retried), we would
+         * be missing a sequence of log entries on disk. The implementation
+         * can't handle that + the accounting of the append index would be off.
+         * Ignore the return value of the function as the entries haven't been
+         * added yet to pending_last_index.
+         */
+        uvAppendFinishPendingRequests(uv, status);
+        /* Rewind size and pending_last_index. */
+        s->size -= s->pending.n;
```

Comment:
TODO -  possibly there's a one off error in here still, investigating before removing draft.